inc/remindertranslation.class.php

```diff
@@ -0,0 +1,340 @@
+                                $params);
+         echo "};";
+         echo "</script>\n";
+
+         echo "<div class='center'>".
+              "<a class='vsubmit' href='javascript:addTranslation".$item->getID()."$rand();'>".
+              __('Add a new translation')."</a></div><br>";
+      }
+
+      $obj   = new self;
+      $found = $obj->find(['reminders_id' => $item->getID()], "language ASC");
+
+      if (count($found) > 0) {
+         if ($canedit) {
+            Html::openMassiveActionsForm('mass'.__CLASS__.$rand);
+            $massiveactionparams = ['container' => 'mass'.__CLASS__.$rand];
+            Html::showMassiveActions($massiveactionparams);
+         }
+
+         Session::initNavigateListItems('ReminderTranslation', __('Entry translations list'));
+
+         echo "<div class='center'>";
+         echo "<table class='tab_cadre_fixehov'><tr class='tab_bg_2'>";
+         echo "<th colspan='4'>".__("List of translations")."</th></tr>";
+         if ($canedit) {
+            echo "<th width='10'>";
+            echo Html::getCheckAllAsCheckbox('mass'.__CLASS__.$rand);
+            echo "</th>";
+         }
+         echo "<th>".__("Language")."</th>";
+         echo "<th>".__("Subject")."</th>";
+         foreach ($found as $data) {
+            echo "<tr class='tab_bg_1'>";
+            if ($canedit) {
+               echo "<td class='center'>";
+               Html::showMassiveActionCheckBox(__CLASS__, $data["id"]);
+               echo "</td>";
+            }
+            echo "<td>";
+            echo Dropdown::getLanguageName($data['language']);
+            echo "</td><td>";
+            if ($canedit) {
+               echo "<a href=\"" . ReminderTranslation::getFormURLWithID($data["id"]) . "\">{$data['name']}</a>";
+            } else {
+               echo  $data["name"];
+            }
+            if (isset($data['answer']) && !empty($data['answer'])) {
+               echo "&nbsp;";
+               Html::showToolTip(Toolbox::unclean_html_cross_side_scripting_deep($data['answer']));
+            }
+            echo "</td></tr>";
+         }
+         echo "</table>";
+         if ($canedit) {
+            $massiveactionparams['ontop'] = false;
+            Html::showMassiveActions($massiveactionparams);
+            Html::closeForm();
+         }
+      } else {
+         echo "<table class='tab_cadre_fixe'><tr class='tab_bg_2'>";
+         echo "<th class='b'>" . __("No translation found")."</th></tr></table>";
+      }
+
+      return true;
+   }
+
+
+   /**
+    * Display translation form
+    *
+    * @param integer $ID
+    * @param array   $options
+    */
+   function showForm($ID = -1, $options = []) {
+      if (isset($options['parent']) && !empty($options['parent'])) {
+         $item = $options['parent'];
+      }
+      if ($ID > 0) {
+         $this->check($ID, READ);
+      } else {
+         // Create item
+         $options['itemtype']         = get_class($item);
```

Comment:
`$item` variable is only initialized if `isset($options['parent']) && !empty($options['parent'])`.

Reply:
You have the same test in lot of showform into GLPI (used for only check right)
Example : dropdowntranslation.class.php or knowbaseitemtranslation.class.php

Reply:
I agree with @cedric-anne. If `$options['parent']` is missing or empty; `$item` won't exists and this will fail. And yes, it is already used in some other places, they are wrong also.
I guess that `parent` is always defined (I di not see yet where the call is done); but since there is a condition, this should be handled. Since this seems mandatory, maybe just a else in the parent condition that display a message and returns false? Or maybe is it possible to "guess" that parent?

Reply:
The test is already exist on ajax/viewsubitem.php so we can delete it.